From: Chandan Rajendra <chandan@linux.vnet.ibm.com>
Newsgroups: gmane.comp.file-systems.btrfs
Subject: [PATCH] Btrfs: btrfs_defrag_file: Fix calculation of max_to_defrag.
Date: Wed,  3 Jun 2015 14:59:54 +0530
Archived-At: <http://permalink.gmane.org/gmane.comp.file-systems.btrfs/45532>

max_to_defrag represents the number of pages to defrag rather than the last
page of the file range to be defragged. Fix this.

Signed-off-by: Chandan Rajendra <chandan@linux.vnet.ibm.com>
---
 fs/btrfs/ioctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index ca5d968..2a45026 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -1318,7 +1318,7 @@ int btrfs_defrag_file(struct inode *inode, struct file *file,
 		i = range->start >> PAGE_CACHE_SHIFT;
 	}
 	if (!max_to_defrag)
-		max_to_defrag = last_index + 1;
+		max_to_defrag = last_index - i + 1;
 
 	/*
 	 * make writeback starts from i, so the defrag range can be
-- 
2.1.0
